packages/holocron/src/loadModule.node.js

```diff
@@ -37,75 +61,288 @@ function fetchModuleContent(moduleUrl) {
     });
 
-  return fetchModuleContentAttempt();
-}
+    if (messages.length > 0) {
+      throw new Error(messages.join('\n'));
+    }
+
+    registerModuleUsingExternals(moduleName);
+  }
+};
+
+/**
+ * Validates Required Externals
+ * @param {object} params
+ * @param {object} params.requiredExternals Required Externals
+ * @param {string} params.moduleName module name
+ * @param {object} params.providedExternals Provided Externals
+ * @param {boolean} params.enableUnlistedExternalFallbacks flag to enable external fallbacks
+ */
+const validateRequiredExternals = ({
+  requiredExternals,
+  moduleName,
+  providedExternals,
+  enableUnlistedExternalFallbacks,
+}) => {
+  const messages = [];
+  let moduleCanBeSafelyLoaded = true;
+  const moduleExternals = {};
+
+  Object.keys(requiredExternals).forEach((externalName) => {
+    const providedExternal = providedExternals[externalName];
+    const requiredExternal = requiredExternals[externalName];
+    const {
+      version, name, integrity, semanticRange,
+    } = requiredExternal;
+    const fallbackExternalAvailable = !!name && !!version;
+    const fallbackBlockedByRootModule = !!providedExternal && !providedExternal.fallbackEnabled;
+
+    if (!providedExternal) {
+      messages.push(`External '${externalName}' is required by ${moduleName}, but is not provided by the root module`);
+      if (!enableUnlistedExternalFallbacks) {
+        moduleCanBeSafelyLoaded = false;
+      }
+    } else if (
+      !validateExternal({
+        providedVersion: providedExternal.version,
+        requestedRange: semanticRange,
+      })
+    ) {
+      messages.push(`${externalName}@${semanticRange} is required by ${moduleName}, but the root module provides ${providedExternal.version}`);
+
+      if (fallbackBlockedByRootModule) {
+        moduleCanBeSafelyLoaded = false;
+      }
+    }
+
+    if (fallbackExternalAvailable) {
+      moduleExternals[name] = {
+        name,
+        version,
+        semanticRange,
+        integrity,
+      };
+    }
+  });
+
+  if (messages.length > 0) {
+    if (moduleCanBeSafelyLoaded || (process.env.ONE_DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS === 'true')) {
+      // eslint-disable-next-line no-console
+      console.warn(messages.join('\n'));
+    } else {
+      throw new Error(messages.join('\n'));
+    }
+
+    setModulesRequiredExternals({ moduleName, externals: moduleExternals });
+  }
+  registerModuleUsingExternals(moduleName);
+};
+
+/**
+ * Fetches Module Config. Returns null if does not exist or an error has occurred
+ * @param {string} baseUrl path to the assets
+ * @returns {Promise<object | null>}
+ */
+const fetchModuleConfig = async (baseUrl) => {
+  const moduleConfigUrl = `${baseUrl}module-config.json`;
 
-function checkStatus(response) {
-  if (response.status >= 200 && response.status < 300) {
-    return response;
+  try {
+    const moduleConfigStr = await fetchAsset(moduleConfigUrl);
+    const moduleConfig = JSON.parse(moduleConfigStr);
```

Comment:
If we know this is going to be JSON, can the `fetchAsset` function be modified to return `.json` with a flag instead of `.text`?